src/site/content/en/blog/file-system-access/index.md

```diff
@@ -319,6 +319,43 @@ await directoryHandle.removeEntry('Abandoned Masterplan.txt');
 await directoryHandle.removeEntry('Old Stuff', { recursive: true });
 ```
 
+### Drag and drop integration
+
+The
+[HTML Drag and Drop interfaces](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API)
+enable web applications to accept
+[dragged and dropped files](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/File_drag_and_drop)
+on a web page.
+During a drag and drop operation, dragged file and directory items are associated
+with file entries and directory entries respectively.
+The ` DataTransferItem.getAsFileSystemHandle()` method returns a `FileSystemFileHandle` object
+if the dragged item is a file, and a `FileSystemDirectoryHandle` object if the dragged item is a directory.
+The listing below shows this in action.
+
+```js/13
+elem.addEventListener('dragover', (e) => {
+  // Prevent navigation.
+  e.preventDefault();
+});
+
+elem.addEventListener('drop', async (e) => {
+  // Prevent navigation.
+  e.preventDefault();
+  // Process all of the items.
+  for (const item of e.dataTransfer.items) {
+    // Careful: `kind` will be 'file' for both file
+    // _and_ directory entries.
+    if (item.kind === 'file') {
+      const entry = await item.getAsFileSystemHandle();
+      if (entry.kind === 'file') {
+        retrun handleFileEntry(entry);
```

Comment:
Why `return` actually? Don't you want to process all items?

Reply:
True, I have changed this to a good old `if else`.